pkg/common/scheduler/statefulset/autoscaler.go

```diff
@@ -72,13 +73,21 @@ func (s *StatefulSetScheduler) autoscale(ctx context.Context) {
 				}
 
 				// Desired ratio is 0.5 (TODO: configurable)
-				scale.Spec.Replicas = int32(math.Ceil(float64(s.usedCapacity(snapshot)+s.pendingVReplicas()) / (float64(s.capacity) * 0.5)))
+				new := int32(math.Ceil(float64(s.usedCapacity(snapshot)+s.pendingVReplicas()) / (float64(s.capacity) * 0.5)))
 
-				s.logger.Infow("updating adapter replicas", zap.Int32("replicas", scale.Spec.Replicas))
+				// Make sure not to scale down past the last pod with placed vpods
+				if new < snapshot.lastOrdinal+1 {
+					new = snapshot.lastOrdinal + 1
+				}
 
-				_, err = s.statefulSetClient.UpdateScale(ctx, s.statefulSetName, scale, metav1.UpdateOptions{})
-				if err != nil {
-					s.logger.Errorw("updating scale subresource failed", zap.Error(err))
+				if new != scale.Spec.Replicas {
```

Comment:
Before this change, we always update the statefulset scale, even when new calculated replica number equals the current replicas. What's the drawback of this always update? Why is this update needed?

Reply:
see comment above.